instruments/abstract_instruments/comm/abstract_comm.py

```diff
@@ -202,7 +203,14 @@ def read(self, size=-1, encoding="utf-8"):
         :return: The read string from the connection
         :rtype: `str`
         """
-        return self.read_raw(size).decode(encoding)
+        if encoding == 'utf-8':
```

Comment:
We should check if encoding is in a list of generic text encodings instead of just restricting to `utf-8`, I'm sure there is a way to do that somehow.

Reply:
Are you referring to character encoding (using non-deterministic prediction with _chardet_, for example) or numeric precision (unpacking)?

Reply:
character encoding

someone may want to specify something other than utf-8

Reply:
I think something like this should handle this case: https://stackoverflow.com/questions/630938/detect-whether-charset-exists-in-python/630974#630974

Where if `codecs.lookup(encoding)` succeeds, then do the usual `.decode(encoding)`, otherwise continue to the logic using `struct.unpack`